src/mapping/rewards.ts

```diff
@@ -0,0 +1,74 @@
+import { Store } from "@subsquid/typeorm-store";
+import { RewardEvent, RewardEventType } from "../model";
+import { Event, ProcessorContext } from "../processor";
+import { Entities, getSs58Address, getContractAddress } from "../utils";
+import { events } from "../types";
+
+export async function handleRewards(
+  event: Event,
+  entities: Entities
+): Promise<void> {
+  console.log("events", JSON.stringify(event));
+
+  switch (event.name) {
+    case events.dappStaking.reward.name:
+      if (events.dappStaking.reward.v1.is(event)) {
+        const { account, era, amount } =
+          events.dappStaking.reward.v1.decode(event);
+        const rewardEvent = new RewardEvent({
```

Comment:
Very similar code repeats 3 times. Refactor this into method. Provide event as param.